database.go

```diff
@@ -145,10 +151,28 @@ func (db *database) Status() error {
 	return nil
 }
 
+// UpdateLag reports the amount of time in between when we expected to run
+// a database update and the current time
+func (db *database) UpdateLag() time.Duration {
+	lag := db.SinceLastUpdate()
+	if lag < db.config.UpdatePeriod {
+		return 0
+	}
+	return lag - db.config.UpdatePeriod
+}
+
+// SinceLastUpdate gives the duration since the last database update
+func (db *database) SinceLastUpdate() time.Duration {
+	db.ml.RLock()
+	defer db.ml.RUnlock()
+
+	return db.config.now().Sub(db.last)
+}
+
 // Update synchronizes the local threat lists with those maintained by the
 // global Safe Browsing API servers. If the update is successful, Status should
 // report a nil error.
-func (db *database) Update(api api) {
+func (db *database) Update(api api) (bool, time.Duration) {
```

Comment:
What do you think about reversing these two return values? It would be symmetric with getting a value from a map, and similar to how `error` as the last argument indicates success/failure.